ocs_ci/ocs/ui/page_objects/buckets_tab.py

```
@@ -647,16 +656,22 @@ def upload_folder_to_bucket(self, folder_path: str, wait_time: int = 2) -> None:
 
         Args:
             bucket_name (str): Name of the bucket to navigate to.
+            provider (str): Storage provider - constants.S3_PROVIDER_NOOBAA or
+                constants.S3_PROVIDER_RGW_INTERNAL.
+
         """
-        self.nav_object_storage_page()
+        obj_storage = self.nav_object_storage_page()
+        if provider != constants.S3_PROVIDER_NOOBAA:
+            obj_storage.select_storage_provider(provider)
         logger.info(f"Navigating to bucket: {bucket_name}")
```

Ruff (0.15.21)

[warning] Logging statement uses f-string

(G004)